### PR TITLE
Add PSP migration blog

### LINKED ISSUE #855: [Enhancement] PSP Migration docs
### Description

It would be nice to have a PSP migration section on how Kyverno compares to PSP and PSA and how to migrate from PSP to Kyverno.

### Slack discussion

_No response_

### PR DESCRIPTION
**Pull request by chipzoller**
## Related issue #

Closes #855

## Proposed Changes

Writes a new blog post on migrating from PSP to Kyverno. If approved, this will be referenced in other areas of the documentation to point to the blog post rather than incorporating directly.

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [x] I have inspected the website preview for accuracy.
- [x] I have signed off my issue.

### CODE REVIEW

**Comment by chipzoller** [comment]
See the deploy preview here:

https://deploy-preview-864--kyverno.netlify.app/blog/2023/05/24/podsecuritypolicy-migration-with-kyverno/

File: content/en/blog/general/psp-migration/index.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+
+**No message customization.** When a PSP blocks a Pod the message can't be customized. Same situation with PSA. The message returned by Kyverno can be fully customized making it very useful to show, for example, a team name or email of someone for developers or users to contact.
+
+**Integrated.** PSP and PSA are both integrated into the control plane. Kyverno is an external component (a dynamic admission controller) and so must be run separately in your cluster. In order to do this, Kyverno uses webhooks.
+
+## Migration Outline
+
+When migrating from PSP to Kyverno, this is the high-level approach that we recommend. Links are provided for the Kyverno concepts which are applicable to this migration, however it is still recommended that before diving into a migration yourself you spend some more time with the documentation and, of course, experimentation in a lab or non-production environment.
```
Comment (JimBugwadia) [inline content/en/blog/general/psp-migration/index.md:57]: Should we recommend scanning the cluster with the CLI as a first step?

```sh
kustomize build https://github.com/kyverno/policies/pod-security/enforce | kubectl kyverno apply --cluster  -
```

**Review by JimBugwadia — COMMENTED** [review]
few comments and suggestions

File: content/en/blog/general/psp-migration/index.md [hunk 2]
```diff
@@ -0,0 +1,366 @@
+---
+date: 2023-05-24
+title: PodSecurityPolicy migration with Kyverno
+linkTitle: PodSecurityPolicy migration with Kyverno
+description: Migrating from PodSecurityPolicy to Kyverno
+draft: false
+---
+
+As you've probably heard or maybe observed at this point in time, [PodSecurityPolicy](https://kubernetes.io/docs/concepts/security/pod-security-policy/) (PSP) in Kubernetes is no more. After a deprecation beginning in v1.21, they were finally removed in v1.25. Many organizations out there are still relying on PSPs and, if you're reading this post, you're probably one of them. As you begin to upgrade your clusters closer and closer towards v1.25, the clock is ticking. The choices with which you are faced are to either delay cluster upgrades, which means you aren't keeping up with the frequent releases and risk being on an unmaintained version, or upgrade to v1.25 or later and simply not have security for Pods. Neither is really an acceptable choice. In this blog post, we'll show you what your options are and provide a step-by-step migration guide for getting off PSP and onto Kyverno so you can feel confident in your ability to upgrade safely.
```
Comment (JimBugwadia) [inline content/en/blog/general/psp-migration/index.md:9]: Consider trimming / shortening sentences (e.g. remove text like "or maybe observed at this point in time,") and breaking into shorter paragraphs for easier reading.

**Comment by chipzoller** [comment]
Pushed fixes and improvements based on comments.

File: content/en/blog/general/psp-migration/index.md (hunk 1 above)
Comment (chipzoller) [reply content/en/blog/general/psp-migration/index.md:63]: It's a good idea. But the output produced from that exact command, even on a small cluster, is going to be overwhelming. Even on a small K3d cluster, the output had over 65 failed results.

How about if we recommended scanning with just the baseline profile using the podSecurity subrule instead? It'll produce a far shorter list and probably a more important one for readers to focus their attention.

```sh
curl -s https://raw.githubusercontent.com/kyverno/policies/main/pod-security/subrule/podsecurity-subrule-baseline/podsecurity-subrule-base
line.yaml | kubectl kyverno apply --cluster -
```

WDYT?

Comment (JimBugwadia) [reply content/en/blog/general/psp-migration/index.md:63]: Yes, scanning with `baseline` first and then progressing to `restricted` makes sense as a recommended approach.

The CLI can also scan one namespace as a time. 

Its missing a nicer report format (SARIF, HTML, etc.), which would make it a lot more usable.

**Review by JimBugwadia — COMMENTED** [review]
answered pending question...

File: content/en/blog/general/psp-migration/index.md (hunk 1 above)
Comment (chipzoller) [reply content/en/blog/general/psp-migration/index.md:65]: Thanks, just pushed this update.

**Review by JimBugwadia — APPROVED** [review]

**Pull request merged**